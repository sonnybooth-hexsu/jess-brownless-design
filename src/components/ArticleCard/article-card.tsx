import React from "react"
import buttons from "../../styles/buttons.module.css"

type ArticleCardProps = {
  title: string
  dateText: string
  logo: string
  excerpt: string
  article: string
  image: string
}

export const ArticleCard = ({
  title,
  dateText,
  excerpt,
  article,
  image,
}: ArticleCardProps) => {
  return (
    <div className="grid grid-cols-12 gap-6">
      <div className="col-span-12 mb-6 lg:col-span-5 lg:col-start-8">
        <img
          className="w-full"
          data-testid="articleItemImage"
          width="360"
          height="240"
          src={image}
          alt="Article advert"
        ></img>
      </div>
      <div className="col-span-12 lg:row-start-1 lg:col-start-1 lg:col-span-6">
        <div className="flex items-center">
          <p className="uppercase overline text-secondaryBrand-9 mb-1">
            {dateText}
          </p>
        </div>
        <h2 className="displayLG">{title}</h2>
        <p className="mt-2 text-gray-500">{excerpt}</p>
        <a
          target="_blank"
          rel="noopener noreferrer"
          className={`inline-block mt-8 ${buttons.btnOutlineDarkSmall}`}
          href={article}
        >
          Read more
        </a>
      </div>
    </div>
  )
}
