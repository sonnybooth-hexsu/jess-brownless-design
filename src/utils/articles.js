import placeholder from "../assets/images/placeholder-image.jpg"
import consultation from "../assets/icons/consultation.svg"

export const Articles = [
  {
    title:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Eget diam.",
    date: "11/03/2020",
    logo: consultation,
    excerpt:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod. Lorem ipsum dolor sit amet.",
    article: "/test-1",
    image: placeholder,
  },
  {
    title:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Eget diam.",
    date: "11/03/2020",
    logo: consultation,
    excerpt:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod. Lorem ipsum dolor sit amet.",
    article: "/test-1",
    image: placeholder,
  },
  {
    title:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Eget diam.",
    date: "11/03/2020",
    logo: consultation,
    excerpt:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod. Lorem ipsum dolor sit amet.",
    article: "/test-1",
    image: placeholder,
  },
  {
    title:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Eget diam.",
    date: "11/03/2020",
    logo: consultation,
    excerpt:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod. Lorem ipsum dolor sit amet.",
    article: "/test-1",
    image: placeholder,
  },
  {
    title:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Eget diam.",
    date: "11/03/2020",
    logo: consultation,
    excerpt:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod. Lorem ipsum dolor sit amet.",
    article: "/test-1",
    image: placeholder,
  },
  {
    title:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Eget diam.",
    date: "11/03/2020",
    logo: consultation,
    excerpt:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod. Lorem ipsum dolor sit amet.",
    article: "/test-1",
    image: placeholder,
  },
]

export default Articles
