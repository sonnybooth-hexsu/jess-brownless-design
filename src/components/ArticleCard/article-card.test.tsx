/* eslint no-undef: 0 */

import React from "react"
import { render, cleanup } from "@testing-library/react"
import { ArticleCard } from "./article-card"
import articles from "../../utils/articles"

afterEach(() => {
  cleanup()
})

describe("<ArticleCard />", () => {
  it("should match snapshot", () => {
    const { container } = render(<ArticleCard {...articles} />)

    expect(container).toMatchSnapshot()
  })

  it("has image item", () => {
    const { queryAllByTestId } = render(<ArticleCard {...articles} />)

    const images = queryAllByTestId("articleItemImage")

    expect(images).toHaveLength(1)
  })
})
