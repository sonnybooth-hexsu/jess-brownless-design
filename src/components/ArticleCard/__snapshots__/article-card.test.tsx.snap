// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleCard /> should match snapshot 1`] = `
<div>
  <div
    class="grid grid-cols-12 gap-6"
  >
    <div
      class="col-span-12 mb-6 lg:col-span-5 lg:col-start-8"
    >
      <img
        alt="Article advert"
        class="w-full"
        data-testid="articleItemImage"
        height="240"
        width="360"
      />
    </div>
    <div
      class="col-span-12 lg:row-start-1 lg:col-start-1 lg:col-span-6"
    >
      <div
        class="flex items-center"
      >
        <p
          class="uppercase overline text-secondaryBrand-9 mb-1"
        />
      </div>
      <h2
        class="displayLG"
      />
      <p
        class="mt-2 text-gray-500"
      />
      <a
        class="inline-block mt-8 btnOutlineDarkSmall"
        rel="noopener noreferrer"
        target="_blank"
      >
        Read more
      </a>
    </div>
  </div>
</div>
`;
